source/javascripts/components/_AddStep.js.erb

```diff
@@ -7,7 +7,19 @@ angular.module("BitriseWorkflowEditor").service("addStepService", function() {
 	var addStepService = {
 		isVisible: false,
 		stepSelected: undefined,
-		steps: undefined
+		steps: undefined,
+		stepIconLoadedStatesByStepIds: {},
+		stepIntersectionObserverCallbacks: {},
+		stepIntersectionObserver: new IntersectionObserver(function(entries) {
```

Comment:
@erosdome what is our browser support guideline? see this: https://caniuse.com/#search=IntersectionObserver it's not supported in many browsers like IE. Should we use a polyfill?